pagerduty/user.go

```diff
@@ -212,6 +224,28 @@ func (s *UserService) Create(user *User) (*User, *Response, error) {
 	return v.User, resp, nil
 }
 
+// findExistingUser searches for a user based on the email
+func (s *UserService) findExistingUser(user *User) (*User, *Response, error) {
+	resp, _, lErr := s.List(&ListUsersOptions{Query: user.Email})
+	if lErr != nil {
+		return nil, nil, fmt.Errorf("[Email has already been taken] but failed to fetch existing users: %w", lErr)
+	}
+
+	for _, u := range resp.Users {
+		if isSameUser(u, user) {
+			return s.Get(u.ID, &GetUserOptions{})
+		}
+	}
+
+	return nil, nil, fmt.Errorf("[Email has already been taken]")
```

Comment:
@Padorax should we just return error here?
```suggestion
	return nil, nil, error)
```